src/plugins/data/common/search/aggs/buckets/lib/histogram_calculate_interval.test.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  calculateHistogramInterval,
+  CalculateHistogramIntervalParams,
+} from './histogram_calculate_interval';
+
+const getLargestPossibleInterval = (params: CalculateHistogramIntervalParams) => {
+  const diff = params.values!.max - params.values!.min;
+  return diff / params.maxBucketsUiSettings;
+};
+
+describe('calculateHistogramInterval', () => {
+  describe('auto calculating mode', () => {
+    let params: CalculateHistogramIntervalParams;
+
+    beforeEach(() => {
+      params = {
+        interval: 'auto',
+        intervalBase: undefined,
+        maxBucketsUiSettings: 100,
+        maxBucketsUserInput: undefined,
+        values: {
+          min: 0,
+          max: 1,
+        },
+      };
+    });
+
+    describe('maxBucketsUserInput is defined', () => {
+      test('should not set interval which more than largest possible', () => {
+        const p = {
+          ...params,
+          maxBucketsUserInput: 200,
+          values: {
+            min: 150,
+            max: 250,
+          },
+        };
+        const expectedInterval = getLargestPossibleInterval(p);
+
+        expect(expectedInterval).toBe(1);
+        expect(calculateHistogramInterval(p)).toBe(expectedInterval);
```

Comment:
I'm not sure I understand why this test has two assertions, instead of:

```suggestion
        expect(calculateHistogramInterval(p)).toEqual(1);
```